annotations/views/rest_views.py

```diff
@@ -476,25 +479,87 @@ def create(self, request, *args, **kwargs):
         return Response(serializer.data,
                         status=status.HTTP_201_CREATED,
                         headers=headers)
-
     @list_route()
     def search(self, request, **kwargs):
         q = request.GET.get('search', None)
         if not q:
-            return Response({'results': []})
-        pos = request.GET.get('pos', None)
-        concepts = goat.Concept.search(q=q, pos=pos, limit=50)
-
-        def _relabel(datum):
-            _fields = {
-                'name': 'label',
-                'id': 'alt_id',
-                'identifier': 'uri'
-            }
-
-            return {_fields.get(k, k): v for k, v in datum.iteritems()}
-        return Response({'results': map(_relabel, [c.data for c in concepts])})
+                return Response({'results': []})
+        force = request.GET.get('force')
 
+        if cache.get(q.lower()) and q and force == None: # If force is true we want to run the query from scratch and not use the cached results
```

Comment:
- q needs to be the first condition or ```q.lower()``` would throw an exception if q is None
-  also: ```and not force```